projects/dxilconv/lib/ShaderBinary/ShaderBinary.cpp

```diff
@@ -297,10 +298,10 @@ void InitInstructionInfo()
 
 void CShaderCodeParser::SetShader(CONST CShaderToken* pBuffer)
 {
-    m_pShaderCode = (CShaderToken*)pBuffer;
-    m_pShaderEndToken = (CShaderToken*)pBuffer + pBuffer[1];
+    m_pShaderCode = const_cast<CShaderToken *>(pBuffer);
+    m_pShaderEndToken = const_cast<CShaderToken *>(pBuffer) + pBuffer[1];
     // First OpCode token
-    m_pCurrentToken = (CShaderToken*)&pBuffer[2];
+    m_pCurrentToken = const_cast<CShaderToken *>(&pBuffer[2]);
```

Comment:
I don't know why these three const_casts are necessary. Since we're assigning to constant pointers, why cast to non constant pointers before that assignment?

Reply:
:( Missed after mark m_pCurrentToken to const.
Fixed.